core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -175,7 +175,7 @@ case class SLit(n: BigInt, w: Width) extends LitArg(n, w) {
   * These are not LitArgs, because not all property literals are integers.
   */
 private[chisel3] case class PropertyLit[T, U](
-  propertyType: PropertyTypeclass[_] { type Underlying = U; type Type = T },
+  propertyType: PropertyTypeclass[T] { type Underlying = U },
```

Comment:
What is this change for? I'm confused by it.

Reply:
I originally removed the Path dependent type of Property because it didn't seem necessary and this was part of that. However, [Albert pointed out](https://github.com/chipsalliance/chisel/pull/3514/files#r1311152427) why it is, in fact, needed so I'm restoring it and backing out this change.